src/sing/stateMachine.ts

```diff
@@ -120,19 +130,37 @@ export class StateMachine<
   }
 
   /**
-   * @param initialState ステートマシンの初期ステート。
-   * @param context ステート間で共有されるコンテキスト。
+   * @param stateFactories ステートのファクトリー関数。
+   * @param context ステートマシンのコンテキスト。
+   * @param initialStateId ステートマシンの初期ステートのID。
    */
   constructor(
     stateFactories: StateFactories<StateDefinitions, Input, Context>,
-    initialState: State<StateDefinitions, Input, Context>,
     context: Context,
+    initialStateId: InitialStateId<StateDefinitions>,
   ) {
     this.stateFactories = stateFactories;
     this.context = context;
 
-    this.currentState = initialState;
+    this.currentState = stateFactories[initialStateId](undefined);
+  }
 
+  /**
+   * ステートの遷移を行う。
+   * このメソッドは副作用として、現在のステートの onExit を呼び出し、遷移後のステートの onEnter を呼び出す。
```

Comment:
副作用と言うかメインの操作な気がしますね！
ちょっと専門用語に自信ないですがこういう感じにするとスリムかも。
（あと実装が変わった時にコメントを変えるの忘れちゃうのを防止できます）
```suggestion
   * ステートを遷移し、ライフサイクルイベントを実行する。
```